### ISSUE TITLE
Allow organizing tours with folders or tags

### ISSUE THREAD

**Issue by jfloresremar** [vsls-contrib/codetour#50]
Thank you for your great extension.
Sometimes we could have multiple tours, which would be good to organize with folders (physical or virtual) or tags with search capabilities

**Comment by gsmile** [comment]
Hi there, thanks for the suggestion! We will look into adding this to our backlog.

**Comment by lostintangent** [comment]
@jfloresremar Thanks for reaching out! Could you explain a little bit more about what you're looking to do? Right now, tour files are written to the `.vscode/tours` folder in your project. Are you interested in adding sub-directories to that folder and having the CodeTour player discover them?

Or are you simply looking to be able to tag a tour, and have the tours be grouped by tag in the `CodeTour` tree? I just ask so we can best understand what you're thinking. Thanks!

**Comment by jfloresremar** [comment]
Hello @lostintangent. At the moment I am gradually developing a large project. For me CodeTour is an excellent tool for practitioners developers to easily integrate into the project and I want to document everything from the beginning with both .md files (docsify) and CodeTour.

I have generated two Tours that I have called "Amplify" and "Status" but both tours are not enough to explain everything related to these topics. When creating tours for each specific topic I will end having an extensive list of tours.
![image](https://user-images.githubusercontent.com/10441071/78461895-73e65880-7692-11ea-8075-9b93c34a10fb.png)

I think this enhancement is related with #44 and i'd like to see tours in a treeview something like this:
![image](https://user-images.githubusercontent.com/10441071/78461911-95474480-7692-11ea-9973-ccef73d9aa58.png)

Thank you!

**Comment by lostintangent** [comment]
Thanks so much for that clarification! In order to create that structure, would you prefer to create actual directories within the “.vscode/tours” folder, and place the respective tour files in them? And then have the tree view reflect that structure? 

Or we’re you thinking about something other solution for authoring this structure? For example, we could enable adding tags to a tour and grouping tours by tag, but if you want an arbitrary deep nesting hierarchy, it might be simpler to just organize tours into sub directories and have the tree view reflect that same hierarchy.

**Comment by jfloresremar** [comment]
> Thanks so much for that clarification! In order to create that structure, what do you think about creating actual directories within the “.vscode/tours” folder, and then placing the respective tour files in them?
Yes, I prefer creating directories (and sub directories) inside ".vscode/tours

About searching i can use the search tool of vscode, just typing "./vscode/tours" inside "files to include"
![image](https://user-images.githubusercontent.com/10441071/78582041-559b6c80-77fa-11ea-988c-1d51e4f53dd0.png)

Thank you!

**Comment by lostintangent** [comment]
@jfloresremar Awesome! Alright, then maybe I'll update this issue title to be focused on enabling folder organization then?

**Comment by hallvard** [comment]
Related to this is the possibility to have a hierarchy within a tour, e.g. I have a tour for a multi-module maven project and have added directory steps for each sub-module followed by steps that explain files within it. It would be great if steps related to that directory/sub-module could be nodes inside the directory step instead. I end up with a very long list that doesn't really show its (natural) hierarchical structure and it's very difficult to skip steps related to that directory/sub-module.

**Comment by tsleal29** [comment]
Hi! We use your extension in a big project and thought it might be useful to have the tours organized by folder. I came here looking for an open issue related to this and found this one so I'll just post it here.

We'd like to contribute to the extension by coding this feature ourselves if this can help! I gave a look at the readme but couldn't manage to test my code by myself. I'd like to know if you'd be able to help us with the first steps and to decide what would be the best approach, as you know the code way better than us, and we try to continue from there.

Please let me know if you can help us! :)
Thanks!

**Comment by tsleal29** [comment]
@lostintangent sorry to tag you but I don't know if someone will see my message as this issue hasn't been active for a while now and I didn't want to open a new one for the same thing.

**Comment by Miko321** [comment]
I'm also looking for this, as it helps organize when there are too many tours.
@lostintangent  There's any chance of implementing this feature?
